### ISSUE TITLE
Process Class files

### ISSUE THREAD

**Issue by IanMayo** [DeepBlueCLtd/LegacyMan#274]
This task is to convert an HTML page for an installation into a DITA class.

Sample input file:
https://github.com/DeepBlueCLtd/LegacyMan/blob/develop/data/Britain_Standalones/britain_1a.html

Sample output file:
https://github.com/DeepBlueCLtd/Fi3ldMan/blob/main/dita/regions/britain/legacy/unit_legacy_a.dita

Note: these elements in the generated `dita` file:
* content goes into `body`, starting with a list of images.  [here](https://github.com/DeepBlueCLtd/LegacyMan/blob/develop/legacyman_parser/parse_images_of_class.py#L41) is code to find relevant images from the page - including the blacklist of images we do not want to process
* two parts of the content come from the table that contains a `<td>` cell with a `colspan` of `6`.   We use that first colspan-6 row as a `divider` for the table. The rows _before_ that colspan become the `summary` block, and the rows after it become the `signatures` block

For the blocks of content extracted, we will ignore the structure, just port it across, converting it to `dita` as necessary.  Note: we'll have to _fix_ any `<a>` anchors in the code - but replacing the href with the current filename.

Elements of source file
* `summary` - first two rows of table with `td` with a `colspan` of `6`
* `signatures` - rows 3 onwards of table with a `td` with `colspan` of `6`
* `propulsion` - contents of a `div` containing an `h1` with word `PROPULSION`
* `remarks` - contents of a `div` containing an `h1` with word `REMARKS`

**Comment by ebenezer-melkamu** [comment]
Hello @IanMayo,
Could you please provide me with some guidance on which function should trigger this feature?

**Comment by IanMayo** [comment]
sure. this is just the "next layer of the onion" - so it's the links in the class listing in your last PR that you had to _mangle_ by pointing to "self".

![image](https://github.com/DeepBlueCLtd/LegacyMan/assets/1108513/13d801ba-c1fd-45ea-a134-f9088595a993)

**Comment by ebenezer-melkamu** [comment]
> sure. this is just the "next layer of the onion" - so it's the links in the class listing in your last PR that you had to _mangle_ by pointing to "self".
> 
> ![image](https://user-images.githubusercontent.com/1108513/257483975-13d801ba-c1fd-45ea-a134-f9088595a993.png)

@IanMayo Got it 👍 I will push a PR when i have something working

**Comment by IanMayo** [comment]
Re-open, there are some remaining bits.

We used `special processing` for the signatures and remarks blocks. But, IMHO we can use a more `generic` processing for `propulsion` and `remarks`.

So, I don't think we can handle the transform by cycling through the content, and handling it bit by bit. I think we need to handle it in this way:
* [ ] identify slug of html
* [ ] take a copy of it (to be sure we can't mangle existing content). So something like:
```
       dita_soup = BeautifulSoup(dita_doctype, 'xml')
       dita_soup.append(propulsion_div)
```
* [ ] do `find_all` for one element to change at a time
* [ ] do an in-place fix to `replace_with` new dita-friendly element
* [ ] append the modified slug to the parent element

I'm pretty sure we can use the above process to handle conversion of these elements:
* table
* a href (hyperlink)
* a id (anchor tag)
* img
* b, i, u

## Special processing for links

We're going to process linked pages in a future issue (#289).   So, look for `<a href="">link</a>` entries, and swap them with a link to the current document, as a placeholder.   Include a `#TODO` in the python to remind us we need to come back and process the file.  The `#TODO` should also refer to a variable that contains the real destination filename.

## Special processing for tables

With colspans,  rowspans, text formatting, tables require some complex processing to move to `DITA` (CALS Format).  So, let's replace `<table>` elements with a marker `[TABLE HERE]`. Include a `#TODO` marker in python to remind us that the table has to be processed.  This issue is resolved in #291
